src/UI/NavigationContainer.elm

```diff
@@ -434,13 +437,37 @@ Clicking on the black layer also activates the closing message.
 -}
 dialog : String -> msg -> Element msg -> Dialog msg
 dialog title onClose body =
-    Dialog
+    Dialog1
         { title = title
         , close = onClose
         , body = body
         }
 
 
+{-|
+
+    `Nav.dialogV2` constructs a [`Nav.Dialog`](#Dialog) from a title and icon.
+    This variant of the dialog does not require you to specify body at the time
+    of creation and you can specify it alongside dialog buttons as an option
+    like this:
+
+    Nav.dialog2 "An example dialog" Icon.warning
```

Comment:
Okay, now I see, this part is wrong.

The idea is to have this dialog a totally separated dialog. So it would do:
```
import UI.V2.Dialog as Dialog
-- ...

container : AppConfig -> Nav.Container msg
container { locale, renderConfig } =
    { title = locale.t "pages.404.title"
    , content = Nav.contentSingle <| view locale renderConfig
    , dialog = Nothing -- Deprecated: use `withDialog` instead
    , hasMenu = False
    }
    |> Nav.containerWithDialog
            (Dialog.dialog "An example dialog"
                     (Element.text "Dialog body")
                     |> Dialog.withIcon Icon.warning
                     |> Dialog.withButtons [submitButton, cancelButton]
             )
```

And when rendering it in showcase, it would be something like:
```elm
            Dialog.dialog "An example dialog"
                     (Element.text "Dialog body")
                     |> Dialog.withIcon Icon.warning
                     |> Dialog.withButtons [submitButton, cancelButton]
                     |> Dialog.renderElement renderConfig
```

I'm adding this back into the RFC...

Reply:
I didn't get why have this, Nav.container already have a way to receive a dialog, having `containerWithDialog` makes things inconsistent in my opinion as you will be able to add dialog in the record and then have the with* function.

Again, we want retrocompatibility, even though we set the old version as deprecated. 

So, IMO makese sense to have:

```elm
    { title = locale.t "pages.404.title"
    , content = Nav.contentSingle <| view locale renderConfig
    , dialog = Just (Nav.dialog blablabla) | Just (Nav.dialogV2 myDialog)
    , hasMenu = False
    }
```

Reply:
Yep, you're right, I've edited my comment here and will do the same in the RFC...